registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/RegistrationStatusServiceImpl.java

```diff
@@ -539,4 +543,84 @@ public Boolean checkUinAvailabilityForRid(String rid) {
+		registrationStatusEntity.setRegistrationType(dto.getRegistrationType());
+		registrationStatusEntity.setReferenceRegistrationId(dto.getReferenceRegistrationId());
+		registrationStatusEntity.setStatusCode(dto.getStatusCode());
+		registrationStatusEntity.setLangCode(dto.getLangCode());
+		registrationStatusEntity.setStatusComment(dto.getStatusComment());
+		registrationStatusEntity.setLatestRegistrationTransactionId(dto.getLatestRegistrationTransactionId());
+		registrationStatusEntity.setIsActive(dto.isActive());
+		registrationStatusEntity.setCreatedBy(dto.getCreatedBy());
+		if (dto.getCreateDateTime() == null) {
+			registrationStatusEntity.setCreateDateTime(LocalDateTime.now(ZoneId.of("UTC")));
+		} else {
+			registrationStatusEntity.setCreateDateTime(dto.getCreateDateTime());
+		}
+		registrationStatusEntity.setUpdatedBy(dto.getUpdatedBy());
+		registrationStatusEntity.setUpdateDateTime(dto.getUpdateDateTime());
+		registrationStatusEntity.setIsDeleted(dto.isDeleted());
+
+		if (registrationStatusEntity.isDeleted() != null && registrationStatusEntity.isDeleted()) {
+			registrationStatusEntity.setDeletedDateTime(LocalDateTime.now(ZoneId.of("UTC")));
+		} else {
+			registrationStatusEntity.setDeletedDateTime(null);
+		}
+
+		registrationStatusEntity.setRetryCount(dto.getRetryCount());
+		registrationStatusEntity.setApplicantType(dto.getApplicantType());
+		registrationStatusEntity.setRegProcessRetryCount(dto.getReProcessRetryCount());
+		registrationStatusEntity.setLatestTransactionStatusCode(dto.getLatestTransactionStatusCode());
+		registrationStatusEntity.setLatestTransactionTypeCode(dto.getLatestTransactionTypeCode());
+		registrationStatusEntity.setRegistrationStageName(dto.getRegistrationStageName());
+		registrationStatusEntity.setLatestTransactionTimes(dto.getLatestTransactionTimes());
+		registrationStatusEntity.setResumeTimeStamp(dto.getResumeTimeStamp());
+		registrationStatusEntity.setDefaultResumeAction(dto.getDefaultResumeAction());
+		return registrationStatusEntity;
+	}
+
+	@Override
+	public void updateRegistrationStatusForWorkFlow(InternalRegistrationStatusDto registrationStatusDto,
+			String moduleId, String moduleName) {
+		regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(),
+				registrationStatusDto.getRegistrationId(),
+				"RegistrationStatusServiceImpl::updateRegistrationStatusForWorkFlow()::entry");
+		boolean isTransactionSuccessful = false;
+		LogDescription description = new LogDescription();
+
+		try {
+			InternalRegistrationStatusDto dto = getRegistrationStatus(registrationStatusDto.getRegistrationId());
+			if (dto != null) {
+				RegistrationStatusEntity entity = convertDtoToEntityForWorkFlow(registrationStatusDto);
+				registrationStatusDao.save(entity);
+				isTransactionSuccessful = true;
+				description.setMessage("Updated registration status successfully");
+			}
+		} catch (DataAccessException | DataAccessLayerException e) {

```

Comment:
The exception handling and audit creation we are already doing in the verticle, is it required to do it again here?

If not required then String moduleId, String moduleName can be avoided in method signature

Reply:
We need exception handling and audit 
in updating record in DB